src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copy = { ...state };
+  const newArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const num in copy) {
+          delete copy[num];
+        };
+        newArr.push({ ...copy });
+        break;
+
+      case 'addProperties':
+        Object.assign(copy, action.extraData);
+        newArr.push({ ...copy });
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete copy[key];
+        }
+        newArr.push({ ...copy });
+    }
```

Comment:
add `default case` to handle errors and show the message if action type is invalid